Netki Wallet Name Service API Quickstart Template

wdawg33 opened the pull request:
The Netki Wallet Name Service lookup API server allows you to quickly integrate the Wallet Name standard into your digital currency platform.

azurecla commented:
Hi __@wdawg33__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

azurecla commented:
Hi __@wdawg33__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

azurecla commented:
Hi __@wdawg33__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,224 @@
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Consider eliminating this to avoid deployment failures. See #944 for guidance

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,224 @@
+      "metadata": {
+        "description": "The location where the Virtual Machine will be deployed."
+      },
+      "allowedValues": [
```
Since VMs can be deployed in almost all locations now, consider removing the allowed values list

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,224 @@
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "dnsNameForPublicIP": {
```
Consider renaming the parameter. See #966

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,224 @@
+    },
+    "ubuntuOSVersion": {
+      "type": "string",
+      "defaultValue": "15.04",
```
I believe 15.04 is removed from Azure image repo. Use 15.10 instead or default to the LTS version.

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
```diff
@@ -0,0 +1,224 @@
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Please make the api version a varaible and reuse across resources. Also move to the GA API 2015-06-15

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
Uri contains an extra folder name of the template

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.json:
Use v1.4 and also add autoUpgradeMinorVersion property

singhkay commented on netki-wns-api-server-on-ubuntu/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,24 @@
+      "value": "userName"
+    },
+    "adminPassword": {
+      "value": "password"
```
GEN-PASSWORD here

wdawg33 commented:
@singhkay Thank you for all of the feedback! We incorporated everything, committed, and pushed. We are still encountering a build failure. In azuredeploy.json, line 207, how should we be referencing our shell script? Originally you mentioned the duplicate folder in the path, but that was actually our branch name. We removed the duplicate but are unsure if you would like for us to include master in the path, our branch, or something else. Thanks!

singhkay commented:
@wdawg33 Yep looks like that's what's causing the failure. Here's an example raw url of another script in the repo. Try basing yours on this

https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/lamp-app/install_lamp.sh

wdawg33 commented:
@singhkay Thanks again for your guidance. We are running npm test locally to try and speed up debugging. We have tried with both of the following URLs (master, branch name from forked repo: netki-wns-api-server-on-ubuntu) and the build is still failing. 

https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh

https://raw.githubusercontent.com/Azure/azure-quickstart-templates/netki-wns-api-server-on-ubuntu/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh

We are stuck. It is not clear to us what URL would actually work at http://arm-validator.40.118.211.57.xip.io/validate, or even what is occurring remotely. We even tried with the URL from our forked repo without success (https://raw.githubusercontent.com/netkicorp/azure-quickstart-templates/netki-wns-api-server-on-ubuntu/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh). Can you provide some insight into how (if at all) the URL is being transposed remotely?

azurecla commented:
__@wdawg33__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

singhkay commented:
@wdawg33 Once you place the Azure link, you won't be able to test locally. For local testing you'll need the actual link to your script file in your fork. We ask you to put the Azure link when you send a PR. our CI replaces the Azure part of the link with your fork. See here https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#rawgithubusercontentcom-links

Also my bad, I see that your PR is not from the master branch. In that case, this is the correct link to put at line 207 in your template

    https://raw.githubusercontent.com/Azure/azure-quickstart-templates/netki-wns-api-server-on-ubuntu/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh

wdawg33 commented:
@singhkay Thank you! We have updated the link but unfortunately the build is still failing. Is the failure still due to the link on line 207?

singhkay commented:
@wdawg33 Deployed locally with this link https://raw.githubusercontent.com/netkicorp/azure-quickstart-templates/netki-wns-api-server-on-ubuntu/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh. The link is fine but the changes you need to make are

* Line 52 - GEN-UNIQUE only works in parameters file since it's not a template language expression. It's only known to our CI. But you can generate unique storage account names as in #944 
* Line 81 - dnsNameForPublicIP needs to be replaced with dnsLabelPrefix

wdawg33 commented:
@singhkay Alright, we are close! I tested locally with the Line 207 link pointing to the forked repo that we have. All tests passed!

✓ netki-wns-api-server-on-ubuntu/azuredeploy.json & netki-wns-api-server-on-ubuntu/azuredeploy.parameters.json should be valid (1666395ms)
  1 passing (28m)
Done, without errors.

I changed the link back to https://raw.githubusercontent.com/Azure/azure-quickstart-templates/netki-wns-api-server-on-ubuntu/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh as suggested, committed, and pushed. The build is still failing.

Is there something else that we should try? Thanks!

singhkay commented:
Looks like Travis is undergoing infra migration and it's messing some things up. I can't even see the log. But I deployed locally and it passed. Merging!

wdawg33 commented:
@singhkay Thank you so much for the merge! We just ran a test on prod and the deployment failed. We noticed that Line 207 is still reflecting the branch name in the path instead of master post merge. Shall we do another pull request and update the link to https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/netki-wns-api-server-on-ubuntu/get-build-run-dockerfile.sh (which works now)

singhkay commented:
@wdawg33 Sure, go ahead

wdawg33 commented:
@singhkay Thank you! We just made the change and created another PR.
